### PR TITLE
ref(spans): Reduce number of tags on span metrics

### PR DESCRIPTION
**Pull request by jjbayer**
During the development of Starfish we've repeatedly added new tags to span metrics without restricting the conditions under which a tag is set (that is, what queries actually require a tag).

This PR introduces a new config for the `exclusive_time` and `exclusive_time_light` metrics, which limits the number of cases in which the metrics and / or their tags are extracted. Some tags are removed entirely.

For now, the new config will only be enabled for projects that have the `all-modules` feature flag enabled, i.e. sentry-internal. After we've verified that the new config covers all required cases, we can enable the new config for all orgs.

ref: https://github.com/getsentry/team-ingest/issues/252

### CODE REVIEW

File: tests/integration/test_store.py [hunk 1]
```diff
@@ -1648,22 +1648,11 @@ def test_span_ingestion(
...
-        {
-            "org_id": 1,
-            "project_id": 42,
-            "name": "d:spans/exclusive_time@millisecond",
-            "type": "d",
-            "value": [500.0, 1500],
-            "timestamp": expected_timestamp,
-            "tags": {"span.status": "ok", "span.op": "default"},
-            "retention_days": 90,
-        },
```
Comment (jjbayer) [inline tests/integration/test_store.py]: No exclusive time for `op:default`.

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap [hunk 2]
```diff
@@ -6,429 +6,13 @@ expression: metrics
...
-            "span.action": "GET",
-            "span.category": "http",
-            "span.description": "GET http://domain.tld",
-            "span.domain": "domain.tld",
-            "span.group": "d9dc18637d441612",
-            "span.op": "http.client",
-            "transaction.method": "POST",
-            "transaction.op": "myop",
-        },
-    },
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:95]: @shruthilayaj @gggritso no more exclusive time metrics for `http.client`, except on mobile screen load spans (see `test_extract_span_metrics_mobile_screen`).

Is there any other use case where we still need `http.client` spans?

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap [hunk 3]
```diff
@@ -637,11 +152,8 @@ expression: metrics
             "span.domain": "table",
             "span.group": "f4a7fef06db3d88e",
             "span.op": "db.sql.query",
-            "span.status": "ok",
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:640]: @gggritso `span.status` will be removed for db spans.

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap [hunk 4]
```diff
@@ -637,11 +152,8 @@ expression: metrics
...
-            "span.system": "postgresql",
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:641]: @gggritso we still use `span.system` in Relay to pick an SQL parser, but it won't be a metric tag anymore.

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap [hunk 5]
```diff
@@ -637,11 +152,8 @@ expression: metrics
...
             "transaction": "gEt /api/:version/users/",
             "transaction.method": "POST",
-            "transaction.op": "myop",
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:644]: @gggritso `transaction.op` will be removed for db spans.

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap [hunk 6]
```diff
@@ -661,10 +173,6 @@ expression: metrics
             "span.domain": "table",
             "span.group": "f4a7fef06db3d88e",
             "span.op": "db.sql.query",
-            "span.status": "ok",
-            "span.system": "postgresql",
-            "transaction.method": "POST",
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:666]: @gggritso `transaction.method` is still a tag on `exclusive_time`, but not on `exclusive_time_light`.

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap [hunk 7]
```diff
@@ -1196,47 +645,6 @@ expression: metrics
...
-            "environment": "fake_environment",
-            "span.category": "cache",
-            "span.description": "GET *",
-            "span.group": "37e3d9fab1ae9162",
-            "span.op": "cache.get_item",
-            "span.status": "ok",
-            "transaction.method": "POST",
-            "transaction.op": "myop",
-        },
-    },
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:1239]: @gggritso no more exclusive time metrics for `cache` spans.

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap [hunk 8]
```diff
@@ -1249,49 +657,6 @@ expression: metrics
...
-                2000.0,
-            ],
-        ),
-        tags: {
-            "environment": "fake_environment",
-            "span.action": "GET",
-            "span.category": "db",
-            "span.description": "GET *",
-            "span.group": "37e3d9fab1ae9162",
-            "span.op": "db.redis",
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:1267]: @gggritso no more exclusive time metrics for redis spans (for now).

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap [hunk 9]
```diff
@@ -1483,13 +754,9 @@ expression: metrics
             "file_extension": "js",
             "span.category": "resource",
             "span.description": "http://domain/static/myscript-*.js",
-            "span.domain": "domain",
             "span.group": "022f81fdf31228bf",
             "span.op": "resource.script",
-            "span.status": "ok",
             "transaction": "gEt /api/:version/users/",
-            "transaction.method": "POST",
-            "transaction.op": "myop",
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:1492]: @DominikB2014 `span.domain`, `span.status`, `transaction.method` and `transaction.op` will disappear for resource spans.

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_mobile.snap [hunk 10]
```diff
@@ -218,7 +221,6 @@ expression: "(&event.value().unwrap().spans, metrics)"
                 "os.name": "iOS",
                 "release": "1.2.3",
                 "span.op": "ui.load.initial_display",
-                "transaction.method": "GET",
```
Comment (jjbayer) [inline relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_mobile.snap:221]: @shruthilayaj `transaction.method` will disappear for mobile span metrics.

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap (hunk 3 above)
Comment (gggritso) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:640]: 👍🏻

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap (hunk 4 above)
Comment (gggritso) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:641]: 👍🏻 makes sense! Though, caveat: there's a chance we'll need this back in the future, when we come back to do a database server selector. Right now we're planning to use the database host (doesn't exist as a tag yet) but we might need the system, too

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap (hunk 5 above)
Comment (gggritso) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:644]: I think this is fine for now, but it's a little risky! When we get back to working on a Web Service View (which hopefully is soon, but I don't know!) we'll probably want to filter time spent to `http.server` spans vs. `task` spans. Again, this is just me speculating, but still!

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap (hunk 6 above)
Comment (gggritso) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:666]: Makes sense, I don't know of any product use-case for querying by transaction method but no transaction

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap (hunk 7 above)
Comment (gggritso) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:1239]: Make sense! Same caveat about Web Service view, we'll probably want `cache` spans then

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap (hunk 8 above)
Comment (gggritso) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:1267]: Make sense! Same caveat about Web Service view, we'll probably want `redis` spans then

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap (hunk 2 above)
Comment (gggritso) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:95]: Not that I know of, but it'll come up immediately when we work on an API module (unsure when, if ever)

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap (hunk 4 above)
Comment (jjbayer) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:641]: Ack, brining it back should not be a problem!

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap (hunk 5 above)
Comment (jjbayer) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:644]: If we can live with not having backdated data at the point where we start developing that view, I would drop it for now and bring it back when it's needed.

File: relay-dynamic-config/src/defaults.rs [hunk 11]
```diff
@@ -47,32 +47,37 @@ pub fn add_span_metrics(project_config: &mut ProjectConfig) {
         return;
     }
 
-    // Add conditions to filter spans if a specific module is enabled.
-    // By default, this will extract all spans.
-    let (span_op_conditions, resource_condition) = if project_config
+    // "all-modules" is our de-facto feature flag for experimental features.
```
Comment (olksdr) [inline relay-dynamic-config/src/defaults.rs]:
```suggestion
    // The "all-modules" is our de-facto feature flag for experimental features.
```

**Review by olksdr — APPROVED** [review]

File: relay-dynamic-config/src/defaults.rs (hunk 11 above)
Comment (jjbayer) [inline relay-dynamic-config/src/defaults.rs]:
```suggestion
    // The "all-modules" feature flag is our de-facto feature flag for experimental features.
```

**Review by iker-barriocanal — APPROVED** [review]

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap (hunk 5 above)
Comment (gggritso) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:644]: 👍🏻

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap (hunk 9 above)
Comment (DominikB2014) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:1492]: @jjbayer We use span.domain for the domain dropdown, will there be an alternative?
image

Comment (jjbayer) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:1492]: Whoops. how did I miss that. Will add it back.

Comment (DominikB2014) [reply relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_all_modules.snap:1492]: 🙏 Thanks @jjbayer!

**Pull request merged**